x/pool-incentives/keeper/grpc_query.go

```diff
@@ -143,52 +142,41 @@ func (q Querier) IncentivizedPools(ctx context.Context, _ *types.QueryIncentiviz
 	// While there are exceptions, typically the number of incentivizedPools
 	// equals to the number of incentivized gauges / number of lockable durations.
 	incentivizedPools := make([]types.IncentivizedPool, 0, len(distrInfo.Records)/len(lockableDurations))
-	incentivizedPoolIDs := make(map[uint64]time.Duration)
 
 	// Loop over the distribution records and fill in the incentivized pools struct.
+recordsLoop:
 	for _, record := range distrInfo.Records {
 		for _, lockableDuration := range lockableDurations {
 			poolId, err := q.Keeper.GetPoolIdFromGaugeId(sdkCtx, record.GaugeId, lockableDuration)
-			if err == nil {
+			// If we cannot find the pool id, it's possible this gauge Id is a group gauge.
+			if err != nil {
+				group, err := q.Keeper.incentivesKeeper.GetGroupByGaugeID(sdkCtx, record.GaugeId)
+				if err != nil {
+					continue
+				}
+				poolIds, durations, err := q.Keeper.incentivesKeeper.GetPoolIdsAndDurationsFromGroup(sdkCtx, group)
+				if err != nil {
+					continue
+				}
+				for i, poolId := range poolIds {
+					incentivizedPool := types.IncentivizedPool{
+						PoolId:           poolId,
+						LockableDuration: durations[i],
+						GaugeId:          record.GaugeId,
+					}
+
+					incentivizedPools = append(incentivizedPools, incentivizedPool)
+				}
+				// no need to loop over the remaining lockableDurations so we move
+				// on to the next record.
+				continue recordsLoop
```

Comment:
Consider simplifying this by reading the gauge itself before the loop:
```
for _, lockableDuration := range lockableDurations {
```

Once read, we can determine its type and handle it appropriately.

If Group -> read Group and its gauge records to get poolID from gauge ID in the record
If NoLock -> grab pool ID directly from gauge
If ByLock -> use the old lockable duration standard

Note, that it might be useful to implement a helper `GetPoolID`FromInternalGaugeID`

Similar to [this](https://github.com/osmosis-labs/osmosis/blob/76fda982cced6dd1cb3708785956b1896f2ba688/x/pool-incentives/keeper/keeper.go#L186) but in the other direction

Reply:
loop simplified here https://github.com/osmosis-labs/osmosis/pull/6601/commits/76e232ffd4c7cf60b56c211f01a9b90a8b6b89f7